.github/workflows/codeql-query.yml

```diff
@@ -165,3 +154,103 @@ jobs:
 
       - name: Combine results
         uses: ./combine-results
+
+  update-repo-tasks-statuses-cancelled:
+    runs-on: ubuntu-latest
+    if: cancelled() && github.event.inputs.variant_analysis_id != ''
+    needs:
+      - run
+    permissions:
+      security-events: write # This permission can be removed soon
+      contents: write
+
+    steps:
+      # To use the action in this repository, we need to check out the repository
+      - name: Checkout codeql-variant-analysis-action repository
+        uses: actions/checkout@v3
+        with:
+          repository: "github/codeql-variant-analysis-action"
+          ref: ${{ github.event.inputs.action_repo_ref }}
+
+      - name: Mask signed auth token
+        if: github.event.inputs.signed_auth_token != ''
+        run: |
+          echo "::add-mask::$(jq --raw-output .inputs.signed_auth_token "$GITHUB_EVENT_PATH")"
+
+      - name: Mask download URL
+        run: |
+          echo "::add-mask::$(jq --raw-output .inputs.instructions_url "$GITHUB_EVENT_PATH")"
+
+      # Download and use ::add::mask for all tokens.
+      # Note that masking does not persist to other/future jobs in the same workflow.
+      - name: Download and mask tokens
+        run: |
+          curl --fail --show-error --location --retry 10 --output instructions.json "$(jq --raw-output .inputs.instructions_url "$GITHUB_EVENT_PATH")"
+          jq '.repositories | [.[].token , .[].pat] | map( select( . != null ) ) | .[]' --raw-output instructions.json | \
+            xargs -I {} echo "::add-mask::{}"
+
+      # Extract the repositories that were analyzed in this workflow from the instructions file.
+      - name: Extract repos
+        id: repos
+        run: |
+          echo "repositories=$(jq --raw-output --compact-output .repositories instructions.json)" >> $GITHUB_OUTPUT
+
+      - name: Handle workflow cancelled
+        uses: ./update-repo-task-status
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          workflow_status: "canceled"
+          controller_repo_id: ${{ github.repository_id }}
+          repositories: ${{ steps.repos.outputs.repositories }}
```

Comment:
I tried to run this on a variant analysis of 1000 repos and this step failed with
```
An error occurred trying to start process '/home/runner/runners/2.299.1/externals/node16/bin/node' with working directory '/home/runner/work/qc-controller/qc-controller'. Argument list too long
```
I think because it's passing the entire repositories input on the command line to the `node` process.

I think we can work around this by having the `update-repo-task-status` action read the inputs directly from the `$GITHUB_EVENT_PATH` file, rather than accepting them from the `with` block in the workflow file.

Reply:
Thanks for noticing this, I've updated the action. However, instead of reading the inputs from the `GITHUB_EVENT_PATH`, we need to read the repositories from the instructions file (to which we only have a URL in the `GITHUB_EVENT_PATH` file). So, the workflow will still download the instructions file and pass the download path to the action, which will read the file and extract the `repositories` from it.